libkbfs/journal_server.go

```diff
@@ -641,6 +652,82 @@ func (j *JournalServer) mdOps() journalMDOps {
 	return journalMDOps{j.delegateMDOps, j}
 }
 
+// estimateQuotaUsage returns quota information, including local usage.
+func (j *JournalServer) estimateQuotaUsage(ctx context.Context) (
+	localUsageBytes, remoteUsageBytes, limitBytes int64) {
+	// Always get a cached quota value, but kick off a background
+	// task to retrieve an updated quota if the cached value is
+	// more than 5s out of date.
+	const quotaUsageBackgroundTolerance = 5 * time.Second
+	timestamp, usageBytes, limitBytes, err :=
+		j.quotaUsage.Get(ctx, quotaUsageBackgroundTolerance,
+			math.MaxInt64)
+	if err != nil {
+		// This shouldn't happen, since we're only retrieving
+		// a cached value. But if it does, just pretend we
+		// have unlimited quota.
+		j.log.CWarningf(ctx,
+			"Error encountered when getting quota usage: %+v", err)
+		return 0, 0, math.MaxInt64
+	} else if timestamp.IsZero() {
+		// If we haven't retrieved a cached value yet, pretend
+		// we have unlimited quota until we get a cached
+		// value.
+		return 0, 0, math.MaxInt64
+	}
+
+	j.lock.RLock()
+	defer j.lock.RUnlock()
+	var totalUnflushedBytes int64
+	for _, tlfJournal := range j.tlfJournals {
```

Comment:
As mentioned in my comment below, the backpressure disk limiter knows about the stored bytes, but that's separate from the unflushed bytes.